akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -95,6 +118,7 @@ sealed trait HttpEntity extends jm.HttpEntity {
 /* An entity that can be used for body parts */
 sealed trait BodyPartEntity extends HttpEntity with jm.BodyPartEntity {
   def withContentType(contentType: ContentType): BodyPartEntity
+  def withSizeLimit(maxBytes: Long): BodyPartEntity
```

Comment:
(doing last round of review right now and let's merge then, sorry this is dragging out so slow @sirthias )